src/hooks/usePerpetualMarketsStats.ts

```diff
@@ -0,0 +1,130 @@
+import { useMemo } from 'react';
+
+import { getChainRevenue } from '@/services';
+import { ResolutionString } from 'public/tradingview/charting_library';
+import { useQueries, useQuery } from 'react-query';
+import { shallowEqual, useSelector } from 'react-redux';
+
+import { getPerpetualMarkets } from '@/state/perpetualsSelectors';
+
+import { log } from '@/lib/telemetry';
+
+import { useDydxClient } from './useDydxClient';
+
+const endDate = new Date();
+const startDate = new Date();
+startDate.setDate(startDate.getDate() - 1);
+
+export const usePerpetualMarketsStats = () => {
+  const perpetualMarkets = useSelector(getPerpetualMarkets, shallowEqual) ?? {};
+  const { getCandles, compositeClient } = useDydxClient();
+
+  const markets = useMemo(
+    () => Object.values(perpetualMarkets).filter(Boolean),
+    [perpetualMarkets]
+  );
+
+  const { data } = useQuery({
+    queryKey: ['chain-revenue', startDate.toISOString(), endDate.toISOString()],
+    queryFn: () => {
+      try {
+        return getChainRevenue({
+          startDate,
+          endDate,
+        });
+      } catch (error) {
+        log('usePerpetualMarketsStats getChainRevenue', error);
+      }
+    },
+    refetchOnWindowFocus: false,
+  });
+
+  const results = useQueries(
+    markets.map((market) => ({
+      enabled: !!compositeClient && markets.length > 0,
+      queryKey: ['perpetualMarketCandles', market.id, '1HOUR'],
+      queryFn: () => {
+        try {
+          return getCandles({
+            marketId: market.id,
+            resolution: '60' as ResolutionString,
+            limit: 24,
+          });
+        } catch (error) {
+          log('usePerpetualMarketsStats getCandles', error);
+        }
+      },
+      refetchOnWindowFocus: false,
+    }))
```

Comment:
@DavideSegullo This candles loop is what will add a huge amount of latency in order to render the volume and OI spark lines. I think there was some miscommunication where you took out the market sparklines instead.

The sparklines for each market are OK.

Reply:
No problem, I'll put back the column with the daily chart. Yes the problem is that we don't have a service that gives me the volume statistic directly so I was calculating it that way. Would you prefer to remove this graph for now?

Reply:
Daily chart is back

Reply:
No problem! It happens, I'll remove the chart if it isn't a UI problem

Reply:
So without `getCandles` we need to remove both open interest and 24h volume charts, we'll need a service for both in the future not just the 24h volume, does it sounds good?

Reply:
Done